ipaplatform/base/services.py

```diff
@@ -298,14 +302,25 @@ def start(self, instance_name="", capture_output=True, wait=True):
             instance_name,
             update_service_list=update_service_list)
 
-    def restart(self, instance_name="", capture_output=True, wait=True):
-        ipautil.run([paths.SYSTEMCTL, "restart",
-                     self.service_instance(instance_name)],
+    def _restart_base(self, instance_name, operation, capture_output=True,
```

Comment:
I think it's better to pass explicit operation to the function to prevent surprise behavior.  The public functions are then more readable.

Feel free to suggest better name for the base function.